apps/wakunode2/wakunode2.nim

```diff
@@ -149,3 +150,30 @@ when isMainModule:
   info "Node setup complete"
 
   runForever()
+
+proc subscribeCallbackToTopic*(pubSubTopic: cstring,
+                               callback: proc(pubsubTopic: string, data: seq[byte]): Future[void] {.gcsafe, raises: [Defect].}) =
+  wakunode2.subscribeCallbackToTopic(pubSubTopic, callback)
+
+proc unsubscribeCallbackFromTopic*(pubSubTopic: cstring,
+                                   callback: proc(pubsubTopic: string, data: seq[byte]): Future[void] {.gcsafe, raises: [Defect].}) =
+  wakunode2.unsubscribeCallbackFromTopic(pubSubTopic, callback)
+
+proc unsubscribeAllCallbacksFromTopic*(pubSubTopic: cstring) =
+  wakunode2.unsubscribeAllCallbackFromTopic(pubSubTopic)
+
+proc publishMessage*(pubSubTopic: cstring, message: WakuMessage): Future[int] {.gcsafe, async.} =
+  return await wakunode2.publishMessage(pubSubTopic, message)
```

Comment:
The way I see it, these do not belong here. The C bindings should be a wrapper around the Nim API defined in `waku_node.nim`. As such the app modules should remain unchanged and preferably unimported elsewhere. Any additional API functionality should go to `waku_node.nim`. Of course, since we need a configured `WakuNode` object for those bindings, we may have to provide `App`-like functionality when initiating the node. For now, though I think the configuration will be much, much simpler than is needed for a `wakunode2` application and providing the basic configuration in a `JsonConfig` via `waku_new()` will be enough.